### PR TITLE
chore: remove tailnet v1 API support

### PR DESCRIPTION
**Pull request by spikecurtis**
Drops support for v1 of the tailnet API, which was the original coordination protocol where we only sent node updates, never marked them lost or disconnected.

v2 of the tailnet API went GA for CLI clients in Coder 2.8.0, so clients older than that would stop working.

### CODE REVIEW

**Comment by spikecurtis** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/14641?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#14641** <a href="https://app.graphite.dev/github/pr/coder/coder/14641?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#14640** <a href="https://app.graphite.dev/github/pr/coder/coder/14640?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @spikecurtis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: enterprise/tailnet/pgcoord_test.go [hunk 1]
```diff
@@ -683,7 +592,13 @@ func TestPGCoordinator_Unhealthy(t *testing.T) {
 	logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
 
 	calls := make(chan struct{})
+	// first call succeeds, so that our Agent will successfully connect.
```
Comment (spikecurtis) [inline enterprise/tailnet/pgcoord_test.go:595]: previously, we started a pipe connection and used the v1 API over it.  This eventually calls `Coordinate()`, but starting the pipe didn't block on `Coordinate()` returning.

With the v2 API, we synchronously call `Coordinate()` from the main test goroutine, so we need to get an initially successful heartbeat to unblock the `Coordinate()` call or this test would deadlock.

**Review by deansheather — APPROVED** [review]
Nice work on the test rewriting 👍

**Comment by spikecurtis** [comment]
### Merge activity

* **Sep 11, 11:40 PM EDT**: @spikecurtis started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/coder/coder/14641).

**Pull request merged**